### PR TITLE
Add CI for crate publishing and `cargo-deny`

### PR DESCRIPTION
**Pull request by dbanty**
The name `quil` is taken, so I've set the new name to `quil-rs`. This is a breaking change for anyone currently consuming the crate as they have to update their `Cargo.toml`.

# TODO

- [x] Add the `CRATES_IO_TOKEN ` secret for release CI

### CODE REVIEW

File: .github/workflows/msrv.yml [hunk 1]
```diff
@@ -99,4 +99,28 @@ jobs:
...
+          toolchain: stable
+          override: true
+
+      - name: Install cargo-deny
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-deny
+
+      - name: Run cargo clippy
```
Comment (genos) [inline .github/workflows/msrv.yml:123]: `name: Run cargo deny`?

File: Cargo.toml [hunk 2]
```diff
@@ -1,9 +1,12 @@
 [package]
-name = "quil"
+name = "quil-rs"
+description = "Rust tooling for Quil (Quantum Instruction Language)"
```
Comment (genos) [inline Cargo.toml:3]: 👏

File: deny.toml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ignore = [
+    #"RUSTSEC-0000-0000",
+]
+
+# This section is considered when running `cargo deny check licenses`
+# More documentation for the licenses section can be found here:
+# https://embarkstudios.github.io/cargo-deny/checks/licenses/cfg.html
+[licenses]
+unlicensed = "deny"
+allow = [
```
Comment (genos) [inline deny.toml:36]: No BSD?

**Review by genos — COMMENTED** [review]
Nicely done! One or two nitpicks, but largely LGTM

File: src/expression.rs [hunk 4]
```diff
@@ -178,7 +178,7 @@ impl Expression {
     /// # Example
     ///
     /// ```rust
-    /// use quil::expression::Expression;
+    /// use quil_rs::expression::Expression;
```
Comment (kalzoo) [inline src/expression.rs:181]: Should this be changed to `crate`?

**Pull request merged**